### PR TITLE
Feature/change registry object hierarchy 531

### PR DESCRIPTION
**Pull request by matthiaszimmermann**

### CODE REVIEW

File: contracts/shared/ComponentService.sol [hunk 1]
```diff
@@ -483,73 +493,55 @@ contract ComponentService is
...
+        // register with registry
+        if (requiredType == PRODUCT()) {
+            componentNftId = _registryService.registerProduct(
+                component, initialOwner).nftId;
+        } else {
+            componentNftId = _registryService.registerProductLinkedComponent(
+                component, requiredType, initialOwner).nftId;
+        }
 
         component.linkToRegisteredNftId();
```
Comment (rapidddenis) [inline contracts/shared/ComponentService.sol:525]: If code after linking can not be executed in a separate tx -> linking MUST be done at the most

File: contracts/shared/ComponentService.sol [hunk 2]
```diff
@@ -626,15 +624,33 @@ contract ComponentService is
...
         }
 
-        // check instance has assigned required role to inital owner
-        instance = _getInstance(info.parentNftId);
-        owner = info.initialOwner;
+        // check component parent nft id matches with sender nft id
+        if (info.parentNftId != senderInfo.nftId) {
+            revert ErrorComponentServiceSenderNotComponentParent(senderInfo.nftId, info.parentNftId);
+        }
+
```
Comment (rapidddenis) [inline contracts/shared/ComponentService.sol:631]: What if instance does not belong to V3 release but still wired to V3 services?
Proposal: check instance version with registry

File: contracts/shared/ComponentVerifyingService.sol [hunk 3]
```diff
@@ -83,7 +83,15 @@ abstract contract ComponentVerifyingService is
             }
         }
 
-        instance = _getInstance(info.parentNftId);
+        // parent of product is instance
```
Comment (rapidddenis) [inline contracts/shared/ComponentVerifyingService.sol:86]: check instance version with registry

File: contracts/shared/ComponentService.sol [hunk 4]
```diff
@@ -585,30 +577,36 @@ contract ComponentService is
...
-    /// @dev based on the provided component address required type and role returns the component and related instance contract
-    /// the function reverts iff:
-    /// - the component has already been registered
+    /// @dev Based on the provided component address required type the component 
+    /// and related instance contract this function reverts iff:
+    /// - the sender is not registered
     /// - the component contract does not support IInstanceLinkedComponent
     /// - the component type does not match with the required type
-    /// - the initial component owner misses the required role (with the instance access manager)
+    /// - the component has already been registered
```
Comment (rapidddenis) [inline contracts/shared/ComponentService.sol:590]: `component.getInitialInfo()` is called twice, here and in registry service.
Potential source of inconsistency?

File: contracts/shared/ComponentService.sol [hunk 5]
```diff
@@ -483,73 +493,55 @@ contract ComponentService is
...
         IComponents.ComponentInfo memory componentInfo = component.getInitialComponentInfo();
+        address tokenAddress = address(componentInfo.token);
         componentInfo.tokenHandler = TokenHandlerDeployerLib.deployTokenHandler(
-            address(componentInfo.token), 
+            tokenAddress, 
             address(instance.getInstanceAdmin().authority()));
 
+        // register component with instance
         instanceStore.createComponent(
             component.getNftId(),
```
Comment (rapidddenis) [inline contracts/shared/ComponentService.sol:541]: use nftId returned by registry

File: contracts/shared/ComponentVerifyingService.sol (hunk 3 above)
Comment (matthiaszimmermann) [reply contracts/shared/ComponentVerifyingService.sol]: instance version checked with _getAndVerifyRegisterableComponent

File: contracts/shared/ComponentService.sol (hunk 5 above)
Comment (matthiaszimmermann) [reply contracts/shared/ComponentService.sol]: fixed

File: contracts/shared/ComponentService.sol (hunk 4 above)
Comment (matthiaszimmermann) [reply contracts/shared/ComponentService.sol:659]: added getRelease to IRegisterable, add checks in componentService

**Pull request merged**